src/server/zset_family.cc

```diff
@@ -222,6 +222,70 @@ OpResult<DbSlice::ItAndUpdater> FindZEntry(const ZParams& zparams, const OpArgs&
   return DbSlice::ItAndUpdater{add_res.it, add_res.exp_it, std::move(add_res.post_updater)};
 }
 
+using RandomPick = std::size_t;
+
+class PicksGenerator {
+ public:
+  virtual RandomPick Generate() = 0;
+  virtual ~PicksGenerator() = default;
+};
```

Comment:
Just two functors in the first place would've done it as well 🙂

Reply:
Can you please explain it in more detail?

Should I remove the `PicksGenerator` interface? If so, then we won't be able to use it like that
`unique_ptr<PicksGenerator> gen = unique ? static_cast<unique_ptr<A>>(make_unique<B>()) : make_unique<C>();`

Reply:
I was just suggesting to use functions as it saves a few dozens of lines

```c++
using Generator = std::function<size_t()>;

Generator NonUniqueGenerator(size_t total_size) {
  return [total_size, bitgen = absl::BitGen{}]() {
    return absl::Uniform(bitgen, 0u, total_size);
  };
}

Generator UniqueGenerator(size_t picks_count, size_t total_size) {
  return ...
}
```

but I don't think it needs to be changed back, this is perferctly fine as well